course/src/isometry.cpp

```diff
@@ -10,9 +10,146 @@
 namespace ekumen {
 namespace math {
 
-/*
- * your code here
- */
+namespace {
+
+Vector3 translation_vector_operation(const Matrix3& matrix,
+                                     const Vector3& first_vector,
+                                     const Vector3& second_vector) {
+  Vector3 aux;
+  aux = matrix * first_vector + second_vector;
+  return aux;
+}
+
+}  // namespace
+
+
+Isometry Isometry::fromTranslation(const Vector3& r_vector) {
+  Isometry aux;
+  aux.translation_vector = r_vector;
+  return aux;
+}
+
+Vector3 Isometry::translation() const {
+  Vector3 aux{translation_vector};
+  return aux;
+}
+
+Vector3 Isometry::transform(const Vector3& r_vector) const {
+  Vector3 aux;
+  aux = (*this) * r_vector;
+  return aux;
+}
+
+Isometry Isometry::inverse() const {
+  double deter = rotation_matrix.det();
+  if (deter == 0) {
+    throw std::invalid_argument("Isometry doesn't have an inverse");
+  }
+  Matrix3 aux;
+  aux = aux.transpose();
+  aux[0][0] = rotation_matrix[1][1] * rotation_matrix[2][2] -
+      rotation_matrix[1][2] * rotation_matrix[2][1];
+  aux[0][1] = rotation_matrix[1][0] * rotation_matrix[2][2] -
+      rotation_matrix[1][2] * rotation_matrix[2][0];
+  aux[0][2] = rotation_matrix[1][0] * rotation_matrix[2][1] -
+      rotation_matrix[1][1] * rotation_matrix[2][0];
+  aux[1][0] = rotation_matrix[0][1] * rotation_matrix[2][2] -
+      rotation_matrix[0][2] * rotation_matrix[2][1];
+  aux[1][1] = rotation_matrix[0][0] * rotation_matrix[2][2] -
+      rotation_matrix[0][2] * rotation_matrix[2][0];
+  aux[1][2] = rotation_matrix[0][0] * rotation_matrix[1][2] -
+      rotation_matrix[0][2] * rotation_matrix[1][0];
+  aux[2][0] = rotation_matrix[0][1] * rotation_matrix[1][2] -
+      rotation_matrix[0][2] * rotation_matrix[1][1];
+  aux[2][1] = rotation_matrix[0][0] * rotation_matrix[1][2] -
+      rotation_matrix[0][2] * rotation_matrix[1][0];
+  aux[2][2] = rotation_matrix[0][0] * rotation_matrix[1][1] -
+      rotation_matrix[0][1] * rotation_matrix[1][0];
+  aux = (aux * Matrix3{
+    1., -1., 1., -1., 1., -1., 1., -1., 1.});
```

Comment:
nit: You could find all the values of the matrix:
```c++
const double c_00 = rotation_matrix[1][1] * rotation_matrix[2][2] -
      rotation_matrix[1][2] * rotation_matrix[2][1];
const double c_01 = rotation_matrix[1][0] * rotation_matrix[2][2] -
      rotation_matrix[1][2] * rotation_matrix[2][0];
...
const double c_22 = rotation_matrix[0][0] * rotation_matrix[1][1] -
      rotation_matrix[0][1] * rotation_matrix[1][0];
```
And then creating the matrix object:
```c++
const Matrix3 m{c_00, c_01, c_02, c_10....c_22};
```

Reply:
In this case since the `transpose()` is the first step, the matrix has to be created before.

Reply:
Check that you obtained the transpose of a zero matrix and then you replace all the values of the matrix. 
So, it is pointless the transpose for two reasons:
 1 - Transpose of zero matrix is zero matrix.
 2 - You didn't use the transposed matrix. You just replaced all the values.

Reply:
Addressed.